src/Analysis/Ast/Impl/Analyzer/Evaluation/ExpressionEval.Collections.cs

```diff
@@ -49,7 +49,7 @@ internal sealed partial class ExpressionEval {
                 }
                 var index = GetValueFromExpression(expr.Index);
                 if (index != null) {
-                    return type.Index(instance, index);
+                    return type.Index(instance, new ArgumentSet(new List<IMember>(){index}, expr, this));
```

Comment:
`new[] { index }`

Generally, we'll make arrays like this (with type inference) rather than making Lists.

Reply:
Got it.